components/mediators/datamapper/org.wso2.carbon.mediator.datamapper/src/main/java/org/wso2/carbon/mediator/datamapper/DataMapperMediator.java

```diff
@@ -282,32 +283,38 @@ private void transform(MessageContext synCtx, String configKey, String inSchemaK
                 JsonUtil.newJsonPayload(axis2MessageContext, outputResult, true, true);
             }
         } catch (ReaderException | InterruptedException | XMLStreamException | SchemaException
-                | IOException | WriterException e) {
+                | IOException | JSException | WriterException e ) {
             handleException("DataMapper mediator : mapping failed", e, synCtx);
         }
     }
 
     private InputStream getInputStream(MessageContext context, String inputType, String inputStartElement) {
         InputStream inputStream = null;
-        switch (InputOutputDataType.fromString(inputType)) {
-        case XML:
-        case CSV:
-            if ("soapenv:Envelope".equals(inputStartElement)) {
+        try {
+            switch (InputOutputDataType.fromString(inputType)) {
+            case XML:
```

Comment:
Thanks for the comment. My code formatter is unable to find the problem. Can you please tell me the issue here.